app/api_topologies.go

```diff
@@ -299,23 +332,22 @@ func renderedForRequest(r *http.Request, topology APITopologyDesc) (render.Rende
 			return render.MakeFilter(render.ComposeFilterFuncs(filters...), renderer)
 		}
 	}
-	return topology.renderer, decorator
+	return topology.renderer, decorator, nil
 }
 
-type reportRenderHandler func(
-	context.Context,
-	Reporter, render.Renderer, render.Decorator,
-	http.ResponseWriter, *http.Request,
-)
+type reportRenderHandler func(context.Context, Reporter, http.ResponseWriter, *http.Request)
 
-func (r *registry) captureRenderer(rep Reporter, f reportRenderHandler) CtxHandlerFunc {
-	return func(ctx context.Context, w http.ResponseWriter, req *http.Request) {
-		topology, ok := r.get(mux.Vars(req)["topology"])
-		if !ok {
-			http.NotFound(w, req)
-			return
-		}
-		renderer, decorator := renderedForRequest(req, topology)
-		f(ctx, rep, renderer, decorator, w, req)
+func (r *registry) rendererForRequest(req *http.Request, rpt report.Report) (render.Renderer, render.Decorator, error) {
+	req.ParseForm()
+	values := map[string]string{}
+	for k, vs := range req.Form {
+		values[k] = vs[0]
+	}
+	return r.rendererForTopology(mux.Vars(req)["topology"], values, rpt)
+}
+
+func captureReporter(rep Reporter, f reportRenderHandler) CtxHandlerFunc {
```

Comment:
(Discussed offline) pls make a captureRender to remove boilerplate from handleTopolog &, handleNode